osquery/events/linux/auditdnetlink.cpp

```diff
@@ -668,6 +675,149 @@ bool AuditdNetlink::configureAuditService() noexcept {
   return true;
 }
 
+bool AuditdNetlink::clearAuditConfiguration() noexcept {
+  int seq = audit_request_rules_list_data(audit_netlink_handle_);
+  if (seq <= 0) {
+    VLOG(1) << "Failed to list the audit rules";
+    return false;
+  }
+
+  // Attempt to list all rules
+  std::vector<AuditRuleDataObject> rule_object_list;
+
+  for (int read_retry = 0; read_retry < 3; ++read_retry) {
```

Comment:
Are we in danger of having an infinite loop here, since `read_retry` can be reset within the loop body?

Mind using a `size_t` if you're only counting.

Reply:
I'm using the same logic used by the auditctl command line; the issue here is that you need to keep reading until you manage to read the reply to what you have requested. I'll add a timeout here!